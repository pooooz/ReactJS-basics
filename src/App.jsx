import './styles/App.css';
import {FuncMessage} from './components/FuncMessage';
import {ClassMessage} from "./components/ClassMessage";

function App() {
  return (
    <div className="App">
      <FuncMessage message="Functional component!"/>
      <ClassMessage message="Hello, class component!"/>
    </div>
  );
}

export default App;
