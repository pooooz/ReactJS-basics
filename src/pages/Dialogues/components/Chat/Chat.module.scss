.dialogue {
  display: flex;
  flex-direction: column;
  align-items: center;
  height: 100%;
  flex-grow: 2;
}

.form {
  margin: 20px 0 0 0;
  display: flex;
}

.form__button {
  font-family: 'Press Start 2P', cursive;
  background: transparent;
  color: limegreen;
  border: none;
  cursor: pointer;
}