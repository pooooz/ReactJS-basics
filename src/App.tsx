import React from 'react';
import './App.css';
import { FuncDialogue } from './components/FuncDialogue/FuncDialogue';
function App() {
  return (
    <div className="App">
      <FuncDialogue />
    </div>
  );
}

export { App };
