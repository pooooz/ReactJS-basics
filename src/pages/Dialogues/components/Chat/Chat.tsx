import React, { useState, FC } from 'react';
import { Button, ThemeProvider } from '@mui/material';
import { createTheme } from '@mui/material/styles';

import { MessageList } from './components/MessageList/MessageList';
import { Input } from './components/Input/Input';

import { MessageInterface } from '../../../../App';

import styles from './Chat.module.scss';

const theme = createTheme({
  typography: {
    fontFamily: 'Press Start 2P',
  },
});

interface FuncDialogueProps {
  addMessage: (a: string) => void;
  messages: MessageInterface[];
}

const MessageListPure = React.memo(MessageList);

export const Chat: FC<FuncDialogueProps> = (props) => {
  const [inputValue, setInputValue] = useState<string>('');

  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
    setInputValue(event.target.value);
  };

  const handleSubmit = (event: React.FormEvent) => {
    if (inputValue) {
      props.addMessage(inputValue);
      setInputValue('');
    }
    event.preventDefault();
  };

  return (
    <ThemeProvider theme={theme}>
      <section className={styles.dialogue}>
        <MessageListPure messages={props.messages} />
        <form className={styles.form} onSubmit={handleSubmit}>
          <Input value={inputValue} change={handleChange} />
          <Button
            className={styles.form__button}
            type="submit"
            style={{
              width: '100px',
              fontFamily: 'Press Start 2P',
              color: '#32CD32FF',
            }}
          >
            Send
          </Button>
        </form>
      </section>
    </ThemeProvider>
  );
};
