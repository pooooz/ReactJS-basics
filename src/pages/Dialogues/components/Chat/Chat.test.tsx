import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import '@testing-library/user-event';
import { Chat } from './Chat';

import { MessageInterface } from '../../../../App';

const initialMessages: MessageInterface[] = [
  {
    id: '1',
    author: 'Test',
    value: 'Test',
  },
];

const addMessage = jest.fn();

describe('Chat', () => {
  it('Render component', () => {
    render(<Chat addMessage={addMessage} messages={initialMessages} />);
  });

  it('Render with snapshot', () => {
    const { asFragment } = render(
      <Chat addMessage={addMessage} messages={initialMessages} />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('Input test', () => {
    render(<Chat addMessage={addMessage} messages={initialMessages} />);
    fireEvent.input(screen.getByRole('textbox'), {
      target: { value: 'Test' },
    });
    expect(screen.getByRole('textbox')).toContainHTML('Test');
  });

  it('Empty message test', () => {
    const { asFragment } = render(
      <Chat addMessage={addMessage} messages={initialMessages} />
    );
    fireEvent.input(screen.getByRole('textbox'), {
      target: { value: '' },
    });
    fireEvent.click(screen.getByRole('button'));
    expect(asFragment()).toMatchSnapshot();
  });

  it('Filled message test', () => {
    render(<Chat addMessage={addMessage} messages={initialMessages} />);
    fireEvent.input(screen.getByRole('textbox'), {
      target: { value: 'Test' },
    });
    fireEvent.click(screen.getByRole('button'));
  });
});
