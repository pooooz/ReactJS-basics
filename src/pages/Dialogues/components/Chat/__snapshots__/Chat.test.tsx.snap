// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chat Empty message test 1`] = `
<DocumentFragment>
  <section>
    <div>
      <div>
        <p
          data-text="Test"
        >
          Test
        </p>
        <p
          data-text="Author: <Test>"
        >
          Author: &lt;Test&gt;
        </p>
      </div>
    </div>
    <form>
      <div>
        <input
          type="text"
          value=""
        />
      </div>
      <button
        class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1t0zgcr-MuiButtonBase-root-MuiButton-root"
        style="width: 100px; font-family: Press Start 2P; color: rgba(50, 205, 50, 1);"
        tabindex="0"
        type="submit"
      >
        Send
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </form>
  </section>
</DocumentFragment>
`;

exports[`Chat Render with snapshot 1`] = `
<DocumentFragment>
  <section>
    <div>
      <div>
        <p
          data-text="Test"
        >
          Test
        </p>
        <p
          data-text="Author: <Test>"
        >
          Author: &lt;Test&gt;
        </p>
      </div>
    </div>
    <form>
      <div>
        <input
          type="text"
          value=""
        />
      </div>
      <button
        class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1t0zgcr-MuiButtonBase-root-MuiButton-root"
        style="width: 100px; font-family: Press Start 2P; color: rgba(50, 205, 50, 1);"
        tabindex="0"
        type="submit"
      >
        Send
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </form>
  </section>
</DocumentFragment>
`;
